typescript/lib/services/service_context.ts

```diff
@@ -0,0 +1,125 @@
+import * as bindingsJsGrpc from '../../kurtosis_core_rpc_api_bindings/api_container_service_grpc_pb'; //TODO - importing all modules or just specific ones
+import * as bindingsJs from '../../kurtosis_core_rpc_api_bindings/api_container_service_pb'; //TODO (comment) - I cannont merge with above line unfortunately, can only import one module at a time MAXIMUM
+import { ServiceID } from './service';
+import { StaticFileID } from './container_creation_config'; 
+import * as grpc from "grpc";
+import * as constructors from "../constructor_calls";
+var path = require("path"); //TODO - I don't if this works
+
+// Docs available at https://docs.kurtosistech.com/kurtosis-libs/lib-documentation
+class GeneratedFileFilepaths { //TODO (comment) - try interface if class doesn't work
+	absoluteFilepathOnTestsuiteContainer: string;
+	absoluteFilepathOnServiceContainer:   string;
+}
+
+class ServiceContext {
+    
+    client: bindingsJsGrpc.ApiContainerServiceClient;
+    serviceId: ServiceID;
+	ipAddress: string;
+	testVolumeMountpointOnTestsuiteContainer: string;
+	testVolumeMountpointOnServiceContainer: string;
+
+    constructor(
+        client: bindingsJsGrpc.ApiContainerServiceClient,
+		serviceId: ServiceID,
+		ipAddress: string,
+		testVolumeMountpointOnTestsuiteContainer: string,
+		testVolumeMountpointOnServiceContainer: string) {
+            this.client = client;
+            this.serviceId = serviceId;
+            this.ipAddress = ipAddress;
+            this.testVolumeMountpointOnTestsuiteContainer = testVolumeMountpointOnTestsuiteContainer;
+            this.testVolumeMountpointOnServiceContainer = testVolumeMountpointOnServiceContainer;
+    }
+
+    // Docs available at https://docs.kurtosistech.com/kurtosis-libs/lib-documentation
+    public getServiceID(): ServiceID { 
+        return this.serviceId;
+    }
+
+    // Docs available at https://docs.kurtosistech.com/kurtosis-libs/lib-documentation
+    public getIPAddress(): string {
+        return this.ipAddress;
+    }
+
+    // Docs available at https://docs.kurtosistech.com/kurtosis-libs/lib-documentation
+    public execCommand(command: string[]): [number, [], Error] { //TODO - [int32, pointer to []byte (comment), error ] ; is is not []`type` like []string
+        let serviceId = this.serviceId;
+        let args = constructors.newGetExecCommandArgs(serviceId, command);
+        let request: grpc.requestCallback<bindingsJs.ExecCommandResponse>;
+
+        let resp, err = this.client.execCommand(args, request); //TODO - does error checking work here ? 
+        
+        if (err != null) {
+            return [0, null, new Error( //TODO - Error type (do I need callback?)
```

Comment:
I am just throwing an error message without throwing the actual error which is a critical thing I need to fix since the user would need an actual error message for debugging purposes. However, I need to figure out if typescript will return an error for me through the typescript bindings or not (I'm trying to see if `grpc.ClientUnaryCall` has some kind of error property that I can use, but it doesn't seem like it does)

Reply:
Oooo yes we definitely want an Error containing a stacktrace

Reply:
I couldn't see a Typescript equivalent to `stacktrace` in Go, though we deffffffinitely want something like that if we can find it